mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/function/DefaultValueResolverFactory.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.runtime.function;
+
+import org.mule.runtime.api.metadata.DataType;
+import org.mule.runtime.api.metadata.FunctionParameter;
+
+/**
+ * Factory definition for {@link FunctionParameter.DefaultValueResolver} resolvers
+ *
+ * @since 1.0
+ */
+@FunctionalInterface
+public interface DefaultValueResolverFactory {
+
+  FunctionParameter.DefaultValueResolver create(Object defaultValue, DataType type);
```

Comment:
WAT?

* If it's Default then why is it an interface?
* Why does it return a DefaultValueResolver instead of a ValueResolver?
* Why is this in ext-api in the first place?